src/libostree/ostree-repo-pull.c

```diff
@@ -2566,6 +2578,103 @@ reinitialize_fetcher (OtPullData *pull_data, const char *remote_name, GError **e
+{
+  g_autofree char *delta_name =
+    _ostree_get_relative_static_delta_superblock_path (from_revision, to_revision);
+  FetchDeltaSuperData *fdata = g_new0(FetchDeltaSuperData, 1);
+  fdata->pull_data = pull_data;
+  fdata->from_revision = g_strdup (from_revision);
+  fdata->to_revision = g_strdup (to_revision);
+
+  _ostree_fetcher_request_to_membuf (pull_data->fetcher,
+                                     pull_data->content_mirrorlist,
+                                     delta_name, 0,
+                                     OSTREE_MAX_METADATA_SIZE,
+                                     0, pull_data->cancellable,
+                                     on_superblock_fetched, fdata);
+  pull_data->n_outstanding_metadata_fetches++;
+  pull_data->n_requested_metadata++;
+}
+
+/* @ref - Optional ref name
+ * @to_revision: Target commit revision we want to fetch
+ *
+ * Start a request for either a ref or a commit.  In the
+ * ref case, we know both the name and the target commit.
+ *
+ * This function primarily handles the semantics around
+ * `disable_static_deltas` and `require_static_deltas`.
+ */
+static gboolean
+initiate_request (OtPullData *pull_data,
+                  const char *ref,
+                  const char *to_revision,
+                  GError    **error)
+{
+  g_autofree char *delta_from_revision = NULL;
+
+  /* Try to use a delta, unless disabled */
+  if (!pull_data->disable_static_deltas)
+    {
+      /* If we have a summary, we can use the newer logic */
+      if (pull_data->summary)
+        {
+          if (!get_best_static_delta_start_for (pull_data, to_revision, &delta_from_revision,
+                                                pull_data->cancellable, error))
+            return FALSE;
+        }
+      else if (ref != NULL)
+        {
+          /* Are we doing a delta via a ref?  In that case we can fall back to the older
+           * logic of just using the current tip of the ref as a delta FROM source. */
+          if (!ostree_repo_resolve_rev (pull_data->repo, ref, TRUE,
+                                        &delta_from_revision, error))
+            return FALSE;
+
+          /* Determine whether the from revision we have is partial; this
+           * can happen if e.g. one uses `ostree pull --commit-metadata-only`.
+           * This mirrors the logic in get_best_static_delta_start_for().
+           */
+          if (delta_from_revision)
+            {
+              OstreeRepoCommitState from_commitstate;
+
+              if (!ostree_repo_load_commit (pull_data->repo, delta_from_revision, NULL,
+                                            &from_commitstate, error))
+                return FALSE;
+
+              /* Was it partial?  Then we can't use it. */
+              if (commitstate_is_partial (pull_data, from_commitstate))
+                g_clear_pointer (&delta_from_revision, g_free);
+            }
+        }
+    }
+
+  /* If we found a delta, let's go */
+  if (delta_from_revision)
```

Comment:
```
+ env LD_PRELOAD=/var/tmp/checkout/tests/libreaddir-rand.so ostree --repo=repo2 pull-local --require-static-deltas repo 37f59afd2f473636ecb96f434cfc5e50a52c7aacf66f6dd765876566b9ced467
error: Static deltas required, but none found for (null) to 37f59afd2f473636ecb96f434cfc5e50a52c7aacf66f6dd765876566b9ced467
ok heuristic endian detection
PASS: tests/test-delta.sh 4 heuristic endian detection
ERROR: tests/test-delta.sh - too few tests run (expected 12, got 4)
ERROR: tests/test-delta.sh - exited with status 1
```

I think this would fix the testsuite as well.